spec/content/automate/ManageIQ/Infrastructure/VM/Retirement/StateMachines/Methods.class/__methods__/check_powered_off_spec.rb

```diff
@@ -0,0 +1,96 @@
+require_domain_file
+
+describe ManageIQ::Automate::Infrastructure::VM::Retirement::StateMachines::CheckPoweredOff do
+  let(:zone) { FactoryGirl.create(:zone) }
+  let(:ems) { FactoryGirl.create(:ems_microsoft, :zone => zone) }
+  let(:vm) do
+    FactoryGirl.create(:vm_microsoft,
+                       :raw_power_state => "PowerOff",
+                       :ems_id          => ems.id)
+  end
+
+  let(:svc_model_vm) do
+    MiqAeMethodService::MiqAeServiceVm.find(vm.id)
+  end
+
+  let(:root_hash) do
+    { 'vm' => MiqAeMethodService::MiqAeServiceVm.find(vm.id) }
+  end
+
+  let(:root_object) do
+    Spec::Support::MiqAeMockObject.new(root_hash)
+  end
+
+  let(:ae_service) do
+    Spec::Support::MiqAeMockService.new(root_object).tap do |service|
+      current_object = Spec::Support::MiqAeMockObject.new
+      current_object.parent = root_object
+      service.object = current_object
+    end
+  end
+
+  it "returns 'ok' for a vm in powered_off state" do
+    described_class.new(ae_service).main
+    expect(ae_service.root['vm'].power_state).to eq("off")
+    expect(ae_service.root['ae_result']).to eq('ok')
+  end
+
+  shared_examples_for "#vm power state" do
+    it "check" do
+      vm.update_attributes(:raw_power_state => raw_power_state)
+      svc_model_vm
```

Comment:
@billfitzgerald0120 
What does this line do? Is it trying to reload the service vm object. Should this be happening after the method has run?